Error with otel

> Downloaded upgraded version distribution-2.10-20260604.065446-264.zip. I am getting an error related to opentelemetry:
> 
> ```
> OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
> [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 1.11.0
> [main] WARN io.opentelemetry.contrib.resourceproviders.AppServerServiceNameProvider - Service name could not be detected using common application server strategies.
> [main] WARN co.elastic.otel.UniversalProfilingProcessorAutoConfig - Cannot start universal profiling integration because no service name was configured
> [main] WARN io.opentelemetry.javaagent.shaded.instrumentation.runtimetelemetry.internal.Internal - otel.instrumentation.runtime-telemetry.emit-experimental-telemetry is deprecated and will be removed in 3.0. Use otel.instrumentation.runtime-telemetry.emit-experimental-metrics instead.
> [main] INFO co.elastic.otel.config.ConfigLoggingAgentListener - AutoConfiguredOpenTelemetrySdk{openTelemetrySdk=OpenTelemetrySdk{tracerProvider=SdkTracerProvider{clock=SystemClock{}, idGenerator=RandomIdGenerator{}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.24.0, attributes={host.arch="amd64", host.name="fermi", os.description="Linux 6.1.0-43-amd64", os.type="linux", os.version="6.1.0-43-amd64", process.command_args=[/usr/lib/jvm/java-17-openjdk-amd64/bin/java, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Dsun.jnu.encoding=UTF-8, -Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager, -Dlog4j.configurationFile=/home/serveracct/fscrawler/config/log4j2.xml, -Dlog4j2.formatMsgNoLookups=true, -javaagent:/home/serveracct/fscrawler/external/elastic-otel-javaagent-1.11.0.jar, -DDOC_LEVEL=debug, -DLOG_LEVEL=debug, -cp, /home/serveracct/fscrawler/external/*:/home/serveracct/fscrawler/lib/*, fr.pilato.elasticsearch.crawler.fs.cli.FsCrawler, --loop, 1], process.executable.path="/usr/lib/jvm/java-17-openjdk-amd64/bin/java", process.pid=8460, process.runtime.description="Debian OpenJDK 64-Bit Server VM 17.0.18+8-Debian-1deb12u1", process.runtime.name="OpenJDK Runtime Environment", process.runtime.version="17.0.18+8-Debian-1deb12u1", service.instance.id="24bb5ddb-37d8-4ef7-94bf-d6688d6b2925", service.name="unknown_service:java", telemetry.distro.name="elastic", telemetry.distro.version="1.11.0", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.61.0"}}, spanLimitsSupplier=SpanLimitsValue{maxNumberOfAttributes=128, maxNumberOfEvents=128, maxNumberOfLinks=128, maxNumberOfAttributesPerEvent=128, maxNumberOfAttributesPerLink=128, maxAttributeValueLength=2147483647}, sampler=DynamicCompositeParentBasedTraceIdRatioBasedSampler(ratio=1.0, io.opentelemetry.contrib.sampler.consistent56.ConsistentParentBasedSampler@3b39e79b), spanProcessor=MultiSpanProcessor{spanProcessorsStart=[io.opentelemetry.javaagent.shaded.instrumentation.thread.internal.AddThreadDetailsSpanProcessor@3ee6dc82], spanProcessorsEnding=[StackTraceSpanProcessor{minSpanDurationNanos=5000000, filterPredicate=co.elastic.otel.SpanStackTraceFilter@2f86f9cf}], spanProcessorsEnd=[co.elastic.otel.common.MutableCompositeSpanProcessor@adafe19, SimpleSpanProcessor{spanExporter=co.elastic.otel.logging.AgentLog$DebugLogSpanExporter@7e2bd5e6, exportUnsampledSpans=false}], spanProcessorsAll=[co.elastic.otel.common.MutableCompositeSpanProcessor@adafe19, SimpleSpanProcessor{spanExporter=co.elastic.otel.logging.AgentLog$DebugLogSpanExporter@7e2bd5e6, exportUnsampledSpans=false}, StackTraceSpanProcessor{minSpanDurationNanos=5000000, filterPredicate=co.elastic.otel.SpanStackTraceFilter@2f86f9cf}, io.opentelemetry.javaagent.shaded.instrumentation.thread.internal.AddThreadDetailsSpanProcessor@3ee6dc82]}, tracerConfigurator=ScopeConfiguratorImpl{conditions=[]}}, meterProvider=SdkMeterProvider{clock=SystemClock{}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.24.0, attributes={host.arch="amd64", host.name="fermi", os.description="Linux 6.1.0-43-amd64", os.type="linux", os.version="6.1.0-43-amd64", process.command_args=[/usr/lib/jvm/java-17-openjdk-amd64/bin/java, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Dsun.jnu.encoding=UTF-8, -Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager, -Dlog4j.configurationFile=/home/serveracct/fscrawler/config/log4j2.xml, -Dlog4j2.formatMsgNoLookups=true, -javaagent:/home/serveracct/fscrawler/external/elastic-otel-javaagent-1.11.0.jar, -DDOC_LEVEL=debug, -DLOG_LEVEL=debug, -cp, /home/serveracct/fscrawler/external/*:/home/serveracct/fscrawler/lib/*, fr.pilato.elasticsearch.crawler.fs.cli.FsCrawler, --loop, 1], process.executable.path="/usr/lib/jvm/java-17-openjdk-amd64/bin/java", process.pid=8460, process.runtime.description="Debian OpenJDK 64-Bit Server VM 17.0.18+8-Debian-1deb12u1", process.runtime.name="OpenJDK Runtime Environment", process.runtime.version="17.0.18+8-Debian-1deb12u1", service.instance.id="24bb5ddb-37d8-4ef7-94bf-d6688d6b2925", service.name="unknown_service:java", telemetry.distro.name="elastic", telemetry.distro.version="1.11.0", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.61.0"}}, metricReaders=[PeriodicMetricReader{exporter=co.elastic.otel.dynamicconfig.BlockableMetricExporter@d504137, intervalNanos=60000000000}], metricProducers=[], views=[], meterConfigurator=ScopeConfiguratorImpl{conditions=[]}}, loggerProvider=SdkLoggerProvider{clock=SystemClock{}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.24.0, attributes={host.arch="amd64", host.name="fermi", os.description="Linux 6.1.0-43-amd64", os.type="linux", os.version="6.1.0-43-amd64", process.command_args=[/usr/lib/jvm/java-17-openjdk-amd64/bin/java, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Dsun.jnu.encoding=UTF-8, -Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager, -Dlog4j.configurationFile=/home/serveracct/fscrawler/config/log4j2.xml, -Dlog4j2.formatMsgNoLookups=true, -javaagent:/home/serveracct/fscrawler/external/elastic-otel-javaagent-1.11.0.jar, -DDOC_LEVEL=debug, -DLOG_LEVEL=debug, -cp, /home/serveracct/fscrawler/external/*:/home/serveracct/fscrawler/lib/*, fr.pilato.elasticsearch.crawler.fs.cli.FsCrawler, --loop, 1], process.executable.path="/usr/lib/jvm/java-17-openjdk-amd64/bin/java", process.pid=8460, process.runtime.description="Debian OpenJDK 64-Bit Server VM 17.0.18+8-Debian-1deb12u1", process.runtime.name="OpenJDK Runtime Environment", process.runtime.version="17.0.18+8-Debian-1deb12u1", service.instance.id="24bb5ddb-37d8-4ef7-94bf-d6688d6b2925", service.name="unknown_service:java", telemetry.distro.name="elastic", telemetry.distro.version="1.11.0", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.61.0"}}, logLimits=LogLimits{maxNumberOfAttributes=128, maxAttributeValueLength=2147483647}, logRecordProcessor=BatchLogRecordProcessor{logRecordExporter=co.elastic.otel.dynamicconfig.BlockableLogRecordExporter@1a17dd6f, scheduleDelayNanos=1000000000, maxExportBatchSize=512, exporterTimeoutNanos=30000000000}, loggerConfigurator=ScopeConfiguratorImpl{conditions=[]}}, propagators=DefaultContextPropagators{textMapPropagator=MultiTextMapPropagator{textMapPropagators=[W3CTraceContextPropagator, W3CBaggagePropagator]}}}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.24.0, attributes={host.arch="amd64", host.name="fermi", os.description="Linux 6.1.0-43-amd64", os.type="linux", os.version="6.1.0-43-amd64", process.command_args=[/usr/lib/jvm/java-17-openjdk-amd64/bin/java, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Dsun.jnu.encoding=UTF-8, -Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager, -Dlog4j.configurationFile=/home/serveracct/fscrawler/config/log4j2.xml, -Dlog4j2.formatMsgNoLookups=true, -javaagent:/home/serveracct/fscrawler/external/elastic-otel-javaagent-1.11.0.jar, -DDOC_LEVEL=debug, -DLOG_LEVEL=debug, -cp, /home/serveracct/fscrawler/external/*:/home/serveracct/fscrawler/lib/*, fr.pilato.elasticsearch.crawler.fs.cli.FsCrawler, --loop, 1], process.executable.path="/usr/lib/jvm/java-17-openjdk-amd64/bin/java", process.pid=8460, process.runtime.description="Debian OpenJDK 64-Bit Server VM 17.0.18+8-Debian-1deb12u1", process.runtime.name="OpenJDK Runtime Environment", process.runtime.version="17.0.18+8-Debian-1deb12u1", service.instance.id="24bb5ddb-37d8-4ef7-94bf-d6688d6b2925", service.name="unknown_service:java", telemetry.distro.name="elastic", telemetry.distro.version="1.11.0", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.61.0"}}, config=io.opentelemetry.sdk.autoconfigure.spi.internal.DefaultConfigProperties@be6d228}
> 
> ```
> Then once fscrawler starts:
> 
> ```
> 11:39:58,113 DEBUG [f.p.e.c.f.t.TikaInstance] Tesseract Path set to [/usr/bin].
> 11:39:58,113 DEBUG [f.p.e.c.f.t.TikaInstance] Tesseract Data Path set to [/usr/share/tesseract-ocr/5/tessdata].
> 11:39:58,123 DEBUG [f.p.e.c.f.t.TikaInstance] OCR strategy for PDF documents is [ocr_and_text] and tesseract was found.
> 11:39:58,123 INFO  [f.p.e.c.f.t.TikaInstance] OCR is enabled. This might slowdown the process.
> 2026-06-08 11:40:05,252 [OkHttp http://localhost:4318/...] ERROR io.opentelemetry.exporter.internal.http.HttpExporter - Failed to export logs. The request could not be executed. Full error message: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:4318
> java.net.ConnectException: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:4318
>         at okhttp3.internal.connection.ConnectPlan.connectSocket(ConnectPlan.kt:278)
>         at okhttp3.internal.connection.ConnectPlan.connectTcp(ConnectPlan.kt:135)
>         at okhttp3.internal.connection.FastFallbackExchangeFinder$launchTcpConnect$1.runOnce(FastFallbackExchangeFinder.kt:141)
>         at okhttp3.internal.concurrent.TaskRunner$runnable$1.run(TaskRunner.kt:81)
>         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
>         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
>         at java.base/java.lang.Thread.run(Thread.java:840)
>         Suppressed: java.net.ConnectException: Failed to connect to localhost/127.0.0.1:4318
>                 ... 7 more
>         Caused by: java.net.ConnectException: Connection refused
>                 at java.base/sun.nio.ch.Net.pollConnect(Native Method)
>                 at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:684)
>                 at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:547)
>                 at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
>                 at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
>                 at java.base/java.net.Socket.connect(Socket.java:633)
>                 at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:148)
>                 at okhttp3.internal.connection.ConnectPlan.connectSocket(ConnectPlan.kt:276)
>                 ... 6 more
> Caused by: java.net.ConnectException: Connection refused
>         at java.base/sun.nio.ch.Net.pollConnect(Native Method)
>         at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:684)
>         at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:547)
>         at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
>         at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
>         at java.base/java.net.Socket.connect(Socket.java:633)
>         at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:148)
>         at okhttp3.internal.connection.ConnectPlan.connectSocket(ConnectPlan.kt:276)
>         ... 6 more
> 2026-06-08 11:40:08,107 [OkHttp http://localhost:4318/...] ERROR io.opentelemetry.exporter.internal.http.HttpExporter - Failed to export spans. The request could not be executed. Full error message: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:4318
> java.net.ConnectException: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:4318
>         at okhttp3.internal.connection.ConnectPlan.connectSocket(ConnectPlan.kt:278)
>         at okhttp3.internal.connection.ConnectPlan.connectTcp(ConnectPlan.kt:135)
>         at okhttp3.internal.connection.FastFallbackExchangeFinder$launchTcpConnect$1.runOnce(FastFallbackExchangeFinder.kt:141)
>         at okhttp3.internal.concurrent.TaskRunner$runnable$1.run(TaskRunner.kt:81)
>         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
>         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
>         at java.base/java.lang.Thread.run(Thread.java:840)
>         Suppressed: java.net.ConnectException: Failed to connect to localhost/127.0.0.1:4318
>                 ... 7 more
>         Caused by: java.net.ConnectException: Connection refused
>                 at java.base/sun.nio.ch.Net.pollConnect(Native Method)
>                 at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:684)
>                 at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:547)
>                 at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
>                 at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
>                 at java.base/java.net.Socket.connect(Socket.java:633)
>                 at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:148)
>                 at okhttp3.internal.connection.ConnectPlan.connectSocket(ConnectPlan.kt:276)
>                 ... 6 more
> Caused by: java.net.ConnectException: Connection refused
>         at java.base/sun.nio.ch.Net.pollConnect(Native Method)
>         at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:684)
>         at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:547)
>         at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
>         at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
>         at java.base/java.net.Socket.connect(Socket.java:633)
>         at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:148)
>         at okhttp3.internal.connection.ConnectPlan.connectSocket(ConnectPlan.kt:276)
>         ... 6 more
> 
> ``` 

 _Originally posted by @ScottCov in [#2395](https://github.com/dadoonet/fscrawler/issues/2395#issuecomment-4650716274)_

--- a/cli/src/main/resources/log4j2.xml
+++ b/cli/src/main/resources/log4j2.xml
@@ -85,6 +85,10 @@
          <AppenderRef ref="Console" />
          <AppenderRef ref="RollingFile" />
       </Logger>
+      <Logger name="io.opentelemetry" level="warn" additivity="false">
+         <AppenderRef ref="Console" />
+         <AppenderRef ref="RollingFile" />
+      </Logger>
 
       <Root level="warn">
          <AppenderRef ref="Console" />

--- a/cli/src/test/resources/log4j2.xml
+++ b/cli/src/test/resources/log4j2.xml
@@ -26,6 +26,9 @@
       <Logger name="org.github.gestalt" level="warn" additivity="false">
          <AppenderRef ref="CONSOLE"/>
       </Logger>
+      <Logger name="io.opentelemetry" level="warn" additivity="false">
+         <AppenderRef ref="CONSOLE"/>
+      </Logger>
       <Root level="info">
          <AppenderRef ref="CONSOLE"/>
       </Root>

--- a/distribution/src/main/scripts/fscrawler
+++ b/distribution/src/main/scripts/fscrawler
@@ -52,13 +52,18 @@ esac
 # Fix for CVE-2021-44228
 JAVA_OPTS="$JAVA_OPTS -Dlog4j2.formatMsgNoLookups=true"
 
-# Auto-load Elastic OTel javaagent if present in external/ (remove the JAR to disable APM tracing)
-# To disable: rm "$FS_HOME/external/elastic-otel-javaagent-*.jar"
+# Always route OTEL agent logs to application logger (Log4j2)
+JAVA_OPTS="$JAVA_OPTS -DOTEL_JAVAAGENT_LOGGING=application"
+
+# Load Elastic OTel javaagent only if OTEL_ENABLED=true
+# If set, will auto-load javaagent if present in external/ (remove the JAR to disable APM tracing)
 # To configure: set OTEL_EXPORTER_OTLP_ENDPOINT, OTEL_SERVICE_NAME, etc.
 # To disable without removing: set OTEL_SDK_DISABLED=true
-OTEL_AGENT=$(ls "$FS_HOME"/external/elastic-otel-javaagent-*.jar 2>/dev/null | head -1)
-if [ -n "$OTEL_AGENT" ]; then
-    JAVA_OPTS="$JAVA_OPTS -javaagent:$OTEL_AGENT"
+if [ "$OTEL_ENABLED" = "true" ]; then
+    OTEL_AGENT=$(ls "$FS_HOME"/external/elastic-otel-javaagent-*.jar 2>/dev/null | head -1)
+    if [ -n "$OTEL_AGENT" ]; then
+        JAVA_OPTS="$JAVA_OPTS -javaagent:$OTEL_AGENT"
+    fi
 fi
 
 # If the user defined FS_JAVA_OPTS, we will use it to start the crawler

--- a/distribution/src/main/scripts/fscrawler.bat
+++ b/distribution/src/main/scripts/fscrawler.bat
@@ -34,18 +34,22 @@ set JAVA_OPTS=%JAVA_OPTS% -Dlog4j.configurationFile=%FS_HOME%/config/log4j2.xml
 REM Fix for CVE-2021-44228
 set JAVA_OPTS=%JAVA_OPTS% -Dlog4j2.formatMsgNoLookups=true
 
-REM Auto-load Elastic OTel javaagent if present in external\ (delete the JAR to disable APM tracing)
-REM To disable without deleting: set OTEL_SDK_DISABLED=true
+REM Always route OTEL agent logs to application logger (Log4j2)
+set JAVA_OPTS=%JAVA_OPTS% -DOTEL_JAVAAGENT_LOGGING=application
+
+REM Load Elastic OTel javaagent only if OTEL_ENABLED=true
 REM PUSHD is guarded so that a missing external\ directory does not unbalance the directory stack.
-IF EXIST "%FS_HOME%\external\" (
-    PUSHD "%FS_HOME%\external"
-    FOR %%I IN (elastic-otel-javaagent-*.jar) DO (
-        set OTEL_AGENT=%FS_HOME%\external\%%I
+IF "%OTEL_ENABLED%"=="true" (
+    IF EXIST "%FS_HOME%\external\" (
+        PUSHD "%FS_HOME%\external"
+        FOR %%I IN (elastic-otel-javaagent-*.jar) DO (
+            set OTEL_AGENT=%FS_HOME%\external\%%I
+        )
+        POPD
     )
-    POPD
-)
-IF DEFINED OTEL_AGENT (
-    set JAVA_OPTS=%JAVA_OPTS% -javaagent:!OTEL_AGENT!
+    IF DEFINED OTEL_AGENT (
+        set JAVA_OPTS=%JAVA_OPTS% -javaagent:!OTEL_AGENT!
+    )
 )
 
 REM If the user defined FS_JAVA_OPTS, we will use it to start the crawler

--- a/docs/source/admin/otel.rst
+++ b/docs/source/admin/otel.rst
@@ -93,8 +93,6 @@ Common variables:
      - Comma-separated ``key=value`` resource attributes
    * - ``OTEL_EXPORTER_OTLP_HEADERS``
      - Auth headers, e.g. ``Authorization=Bearer <token>``
-   * - ``OTEL_SDK_DISABLED``
-     - Set to ``true`` to disable without removing the agent
    * - ``OTEL_EXPORTER_OTLP_TIMEOUT``
      - Export timeout in ms (e.g. ``1000``); reduce if the collector is unavailable
    * - ``OTEL_METRIC_EXPORT_INTERVAL``
@@ -122,18 +120,18 @@ You can find your OTel endpoint and API key in Kibana under
 Behavior without a collector
 ------------------------------
 
-If the agent is present but no EDOT Collector is reachable, the OTLP exporter
+If OTEL_ENABLED is set and the collector is unreachable, the OTLP exporter
 will log a ``WARN`` message on each failed export attempt and retry with
 exponential back-off.  FSCrawler continues to run normally — tracing failures
 are non-blocking.
 
-To suppress the warnings when no collector is available::
+To suppress the warnings, you can:
 
-    export OTEL_SDK_DISABLED=true
-
-or reduce the export timeout::
+- Reduce the export timeout::
 
     export OTEL_EXPORTER_OTLP_TIMEOUT=1000
+
+- Or disable OTEL tracing entirely by not setting ``OTEL_ENABLED=true``
 
 Using with other OTel backends (Jaeger, Zipkin, …)
 ----------------------------------------------------

--- a/docs/source/admin/otel.rst
+++ b/docs/source/admin/otel.rst
@@ -70,6 +70,7 @@ Use standard OpenTelemetry environment variables before starting FSCrawler:
 .. code-block:: bash
 
    # Send traces to an EDOT Collector (OTLP HTTP)
+   export OTEL_ENABLED=true
    export OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4318
    export OTEL_SERVICE_NAME=fscrawler
    export OTEL_RESOURCE_ATTRIBUTES=deployment.environment=production,service.version=2.10

--- a/docs/source/admin/otel.rst
+++ b/docs/source/admin/otel.rst
@@ -11,9 +11,8 @@ loads it on startup and exports traces to any OpenTelemetry-compatible backend
 
 .. note::
 
-   OTel tracing is **enabled by default** when using the standard FSCrawler distribution.
-   The agent JAR is bundled in ``external/``.  No configuration is required to
-   start collecting traces — just point the exporter at your EDOT Collector.
+   OTel tracing is **disabled by default** when using the standard FSCrawler distribution.
+   To enable it, set the ``OTEL_ENABLED=true`` environment variable before starting FSCrawler.
 
 How it works
 -------------
@@ -51,16 +50,16 @@ FSCrawler uses a hybrid instrumentation approach:
      - ``es.bulk.actions``
      - Elasticsearch bulk indexing request (number of operations in the batch)
 
-Disabling OTel tracing
------------------------
+Enabling OTel tracing
+----------------------
 
-Remove the agent JAR to disable tracing entirely::
+To enable OTel tracing, set the ``OTEL_ENABLED=true`` environment variable before starting FSCrawler::
 
-    rm $FS_HOME/external/elastic-otel-javaagent-*.jar
+    export OTEL_ENABLED=true
+    export OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4318
+    export OTEL_SERVICE_NAME=fscrawler
+    export OTEL_RESOURCE_ATTRIBUTES=deployment.environment=production,service.version=2.10
 
-Alternatively, keep the JAR but disable the SDK at runtime::
-
-    export OTEL_SDK_DISABLED=true
     ./bin/fscrawler
 
 Configuring the OTel exporter
@@ -110,6 +109,7 @@ Using with Elastic Cloud (managed EDOT)
 
 .. code-block:: bash
 
+   export OTEL_ENABLED=true
    export OTEL_EXPORTER_OTLP_ENDPOINT=https://<your-otel-endpoint>:443
    export OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer <api-key>"
    export OTEL_SERVICE_NAME=fscrawler
@@ -142,6 +142,7 @@ Example for Jaeger (OTLP/gRPC):
 
 .. code-block:: bash
 
+   export OTEL_ENABLED=true
    export OTEL_EXPORTER_OTLP_ENDPOINT=http://jaeger-host:4317
    export OTEL_SERVICE_NAME=fscrawler
    ./bin/fscrawler
@@ -158,21 +159,20 @@ is available under ``contrib/docker-compose-example-edot/``.
      fscrawler:
        image: dadoonet/fscrawler:latest
        environment:
+         - OTEL_ENABLED=true
          - OTEL_EXPORTER_OTLP_ENDPOINT=http://edot-collector:4318
          - OTEL_SERVICE_NAME=fscrawler
          - OTEL_RESOURCE_ATTRIBUTES=deployment.environment=docker
 
-To disable OTel tracing in Docker without rebuilding the image::
-
-    environment:
-      - OTEL_SDK_DISABLED=true
+To disable OTel tracing in Docker, simply omit the ``OTEL_ENABLED=true`` environment variable.
 
 Windows
 --------
 
-The ``bin\fscrawler.bat`` launcher applies the same auto-detection logic.
+The ``bin\fscrawler.bat`` launcher applies the same logic.
 Set environment variables in the same shell before running::
 
+    set OTEL_ENABLED=true
     set OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4318
     set OTEL_SERVICE_NAME=fscrawler
     bin\fscrawler.bat

